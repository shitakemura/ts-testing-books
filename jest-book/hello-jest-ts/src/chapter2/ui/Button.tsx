import { useState } from 'react'
import './button.css'

type ButtonProps = {
  primary?: boolean
}

export const Button = ({
  primary = true,
}: ButtonProps) => {
  const [isToggleOn, setIsToggleOn] = useState(true)

  const className = primary ? 'primary' : 'secondary'

  return (
    <button
      className={className}
      onClick={() => setIsToggleOn((prev) => !prev)}
    >
      {isToggleOn ? 'ON' : 'OFF'}
    </button>
  )
}
