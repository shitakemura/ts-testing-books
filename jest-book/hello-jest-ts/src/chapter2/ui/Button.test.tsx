/**
 * @jest-environment jsdom
 */

import renderer from 'react-test-renderer'
import {
  render,
  fireEvent,
  queryByText,
} from '@testing-library/react'
import { Button } from './Button'

describe('UI snapshot test', () => {
  // snapshot test with react-test-renderer
  test('renders correctly with react-test-renderer', () => {
    const button = renderer.create(<Button />)
    expect(button.toJSON()).toMatchSnapshot()
  })
})

describe('UI interaction test', () => {
  test('changes the button text upon clicking the button with React Testing Library', () => {
    const button = render(<Button />)

    fireEvent.click(button.getByText('ON'))
    expect(button.getByText(/OFF/i)).toBeTruthy()

    fireEvent.click(button.getByText('OFF'))
    expect(button.getByText(/ON/i)).toBeTruthy()
  })
})
