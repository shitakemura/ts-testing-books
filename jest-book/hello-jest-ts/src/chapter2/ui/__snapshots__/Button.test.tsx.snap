// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI snapshot test renders correctly with react-test-renderer 1`] = `
<button
  className="primary"
  onClick={[Function]}
>
  ON
</button>
`;
